memory/allocator.go

```diff
@@ -26,6 +28,13 @@ type Allocator interface {
 	Account(size int) error
 }
 
+type StatsAllocator interface {
```

Comment:
This stats allocator interface isn't necessary. Anywhere that needs stats should have direct access or be the owner of the resource allocator and should keep a reference to it. It's a bit of an anti-pattern to define an interface to forward method calls down the chain if we can avoid it.